README_developers.md

```diff
@@ -91,128 +85,356 @@ $ # For a fix
 ```
 
 ## Naming Conventions
 
-- Classes
-  - Pascal case (upper and lower case, no underscores)
-- Data members
-  - Snake case
-    - lower case only
-    - use _ to separate
-  - Ending in _
-- Member methods
-  - Snake case
-    - lower case only
-    - use _ to separate
-- Avoid encoding type information in names
+### Classes
 
-For example:
+Use Pascal case for class names (upper and lower case, no underscores).
+When the name is composite of multiple words, each word starts with a capital
+letter. If one of the "words" is an acronym use same capitalization (first
+capital and other letters lowercase).
+```c++
+class MyHiopClass;
+```
+
+### Variables
+
+Use "snake case" for all variables. All lower case; for composite names
+separate words with underscores.
+```c++
+int my_int_variable;
+```
+
+### Member variables
+
+All member variable names should end with trailing underscore to distingusish
+them from local variables.
+```c++
+int memeber_variable_; // Yes!
+
+int another_member_var;      // No!
+int _yet_another_member_var; // No!
+```
+### Function names
+
+Use the snake case for function names, as well.
+separate words with underscores.
+```c++
+void my_hiop_function(int i);
+```
+
+Avoid encoding type information in names and use function
+overloading instead. For example:
 ```cpp
-// Bad, prefer polymorphism
+// No
 void print_double(double val);
 void print_str(std::string_view val);
 
-// Good
+// Yes
 void print(double val);
 void print(std::string_view val);
+```
+
+## Declarations
+
+### Variables
+
+Have each variable declaration have its own type declaration
+For example:
 
-// Good
-class FooBar
+```cpp
+// Yes
+int a;
+int b;
+int c;
+
+// No
+int a, b, c;
+```
+
+### Long declaration lines
+When breaking methods declaration into multiple lines:
+- have a limit of ~125 characters per line
+- keep the first argument on the same line with the method name
+- align arguments on the subsequent lines with the first argument
+- if class name or return type long, place the function name at the beginning of the line and put the return type and class name qualification on the previous line (see example below)
+
+```c++
+// Yes: all arguments on the same line
+void ClassName::some_method(double beta, double alpha) const
 {
-private:
-  double* data_device_;
-public:
-  void to_device();
 }
 
-// Bad
-class Foo_bar {
-  private:
-    double* dataDevice;
-  public:
-    void toDevice();
+// Yes: each argument on a separate line
+void ClassName::some_method(double beta,
+                            hiopMatrix& W,
+                            double alpha,
+                            const hiopMatrix& X) const
+{
+}
+
+// Yes: when return type + class name + method name + first argument would overrun the 125 characters line limit, have method name on the next line
+long_return_type SomeVeryLongClassNameIMeanLong::
+some_method(double beta,
+            hiopMatrix& W,
+            double alpha,
+            const hiopMatrix& X) const
```

Comment:
Do we have a policy for inheritance lists and constructor initialization of member variables? These tend to run long as well. I recommend leading commas (kind of a boost-ism) if you feel it's worthwhile to mention.

Reply:
I have the same question. 
hiopAverylong::
hiopAverylong(typeA a,
                       typeB b)
: nameacanbeverylong_(a),b_(b)

Reply:
i've update the guideline, take a look and let me know